atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClient.java

```diff
@@ -66,7 +69,22 @@ public CassandraClient(Cassandra.Client delegate, AtlasDbQosClient qosClient) {
     public Map<ByteBuffer, List<ColumnOrSuperColumn>> multiget_slice(List<ByteBuffer> keys, ColumnParent column_parent,
             SlicePredicate predicate, ConsistencyLevel consistency_level)
             throws InvalidRequestException, UnavailableException, TimedOutException, TException {
-        return checkLimitAndCall(() -> super.multiget_slice(keys, column_parent, predicate, consistency_level));
+        return checkLimitAndCall(() -> {
+            Map<ByteBuffer, List<ColumnOrSuperColumn>> result = super.multiget_slice(keys, column_parent,
+                    predicate, consistency_level);
+
+            qosMetrics.updateReadCount();
+            qosMetrics.updateBytesRead(getApproximateReadByteCount(result));
+            return result;
+        });
+    }
+
+    private int getApproximateReadByteCount(Map<ByteBuffer, List<ColumnOrSuperColumn>> result) {
+        return result.entrySet().stream()
+                .mapToInt((entry) -> entry.getKey().array().length + entry.getValue().stream()
```

Comment:
this is hard to reason about. Perhaps:
`.mapToInt(entry -> getRowKeySize(entry) + getValueSize(entry))`

Also noting that we briefly discussed verbally whether the computational overhead of this is low enough. We think so.